src/Nox.Types/Types/DateTime/DateTime.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Nox.Types;
+public sealed class DateTime : ValueObject<System.DateTime, DateTime>
+{
+    public int Year
+    {
+        get { return Value.Year; }
+        private set { Value = new System.DateTime(value, Value.Month, Value.Day, Value.Hour, Value.Minute, Value.Second); }
+    }
+    public int Month
+    {
+        get { return Value.Month; }
+        private set => Value = new System.DateTime(Value.Year, value, Value.Day, Value.Hour, Value.Minute, Value.Second);
+    }
+    public int Day
+    {
+        get { return Value.Day; }
+        private set { Value = new System.DateTime(Value.Year, Value.Month, value, Value.Hour, Value.Minute, Value.Second); }
+    }
+    public int Hour
+    {
+        get { return Value.Hour; }
+        private set { Value = new System.DateTime(Value.Year, Value.Month, Value.Day, value, Value.Minute, Value.Second); }
+    }
+    public int Minute
+    {
+        get { return Value.Minute; }
+        private set { Value = new System.DateTime(Value.Year, Value.Month, Value.Day, Value.Hour, value, Value.Second); }
+    }
+    public int Second
+    {
+        get { return Value.Second; }
+        private set { Value = new System.DateTime(Value.Year, Value.Month, Value.Day, Value.Hour, Value.Minute, value); }
+    }
+
+    public DateTime() { Value = System.DateTime.MinValue; }
+
+    public static DateTime From(int year, int month, int day, int hour = 0, int minute = 0, int second = 0) => From(new YearTypeOptions(), year, month, day, hour, minute, second);
+
+
+    public static DateTime From(YearTypeOptions options, int year, int month, int day, int hour = 0, int minute = 0, int second = 0)
```

Comment:
The YearTypeOptions should only be used by Year Type is not designed to be used in DateTime

We need to define a good DateTimeOptions (Min. Max) and build a proper Validation without using the other types, should use .Net Datetime only

Lets keep this very simple